wallet/wallet-rpcd/src/main.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod cmdline;
+
+use logging::log;
+use wallet_rpc_lib::config::{WalletRpcConfig, WalletServiceConfig};
+
+#[tokio::main]
+async fn main() {
+    utils::rust_backtrace::enable();
+
+    if std::env::var("RUST_LOG").is_err() {
+        std::env::set_var("RUST_LOG", "info");
+    }
+
+    logging::init_logging();
+
+    let cmdline::Args {
+        wallet_file,
+        wallet_rpc_address,
+        chain_type,
+        node_rpc_address,
+        node_cookie_file_path,
+    } = clap::Parser::parse();
+
+    let wallet_service_config = WalletServiceConfig {
+        chain_type,
+        wallet_file,
+        node_rpc_address,
+        node_cookie_file_path,
+    };
+
+    let rpc_config = WalletRpcConfig {
+        bind_addr: wallet_rpc_address.as_deref().unwrap_or("127.0.0.1:3333").parse().unwrap(),
```

Comment:
Please use the same way we choose the default node RPC port. Having a fixed port for everything can cause conflicts between different network types. The right way is that for mainnet it can be 3333, for testnet 13333, for regtest 23333, etc. Check ChainConfig.

Reply:
This is preliminary, should have put a TODO there. Thanks for hints of how to go about it.